### PR TITLE
Added network kill

### LINKED ISSUE #854: Write automatic kill system
While testing on Sunday we realized that when the boat loses connection while in RC the motors keep their last control. To solve this we need an automatic kill switch that turns the motors off when no signal from the base station is detected.

### PR DESCRIPTION
**Pull request by andrew-aj**
Closes #854

### CODE REVIEW

**Review by kawaiiPlat — APPROVED** [review]
ye

**Review by kawaiiPlat — CHANGES_REQUESTED** [review]
Gives a timeout during unit testing.

File: NaviGator/hardware_drivers/navigator_kill_board/nodes/kill_board_driver.py [hunk 1]
```diff
@@ -89,12 +92,19 @@ def __init__(self):
...
         self._hw_kill_listener.wait_for_server()
         self._kill_listener.wait_for_server()
+        self._network_kill_listener.wait_for_server()
         rospy.Subscriber("/wrench/selected", String, self.wrench_cb)
-        rospy.Subscriber(
-            "/network", Header, self.network_cb
-        )  # Passes along network hearbeat to kill board
+        # rospy.Subscriber(
+        #     "/network", Header, self.network_cb
+        # )  # Passes along network hearbeat to kill board
```
Comment (cbrxyz) [inline NaviGator/hardware_drivers/navigator_kill_board/nodes/kill_board_driver.py:106]: This should be removed or a comment should be added explaining why this is commented out.

File: NaviGator/hardware_drivers/navigator_kill_board/nodes/kill_board_driver.py [hunk 2]
```diff
@@ -343,21 +372,28 @@ def request(self, write_str, expected_response=None):
         Returns True or False depending on the response.
         With no `recv_str` passed in the raw result will be returned.
         """
-        self.ser.write(write_str)
+        self.ser.write(bytes(write_str, "utf-8"))
```
Comment (cbrxyz) [inline NaviGator/hardware_drivers/navigator_kill_board/nodes/kill_board_driver.py:375]: 22f3d47 needs to be merged in to this branch.

File: NaviGator/hardware_drivers/navigator_kill_board/nodes/kill_board_driver.py [hunk 3]
```diff
@@ -267,12 +278,30 @@ def wrench_cb(self, msg):
...
-    def network_cb(self, msg):
+    def network_kill_alarm_cb(self, alarm):
         """
         Pings kill board on every network hearbeat message. Pretends to be the rf-based hearbeat because
         real one does not work :(
         """
-        self.request(constants["PING"]["REQUEST"], constants["PING"]["RESPONSE"])
+        if alarm.raised:
+            self.nw_killed = True
+            if self.sw_killed:
```
Comment (cbrxyz) [inline NaviGator/hardware_drivers/navigator_kill_board/nodes/kill_board_driver.py:288]: Unless I'm missing something, we could maybe expand `nw_killed` and `sw_killed` to their full name. I'm having a hard time understanding what the "nw" and "sw" mean.

File: NaviGator/hardware_drivers/navigator_kill_board/test/kill_board_test_4.py [hunk 4]
```diff
@@ -15,17 +17,23 @@
...
-        self.updated = False
+        self.network_kill_alarm = None
+        self.hardware_updated = False
+        self.network_updated = False
         self.AlarmListener = AlarmListener("hw-kill", self._hw_kill_cb)
+        self.NetworkListener = AlarmListener("network-loss", self._network_kill_cb)
         self.AlarmBroadcaster = AlarmBroadcaster("kill")
+        self.network = NetworkLoss()
         super().__init__(*args)
+        # self.heartbeat = HeartbeatServer("/network", 1.0)
```
Comment (cbrxyz) [inline NaviGator/hardware_drivers/navigator_kill_board/test/kill_board_test_4.py:28]: Why is this commented out?

File: NaviGator/hardware_drivers/navigator_kill_board/test/kill_board_test_4.py [hunk 5]
```diff
@@ -34,98 +42,62 @@ def _hw_kill_cb(self, alarm):
...
+        hardware_pass = alarm[0].raised if alarm[0] is not None else True
+        network_pass = alarm[1].raised if alarm[1] is not None else True
+        self.assertEqual(hardware_pass and network_pass, False)
 
     def test_4_remote(self):
         """
         Tests remote kill by publishing hearbeat, stopping and checking alarm is raised, then
         publishing hearbeat again to ensure alarm gets cleared.
         """
+        # self.heartbeat = HeartbeatServer("/network", 0.5)
```
Comment (cbrxyz) [inline NaviGator/hardware_drivers/navigator_kill_board/test/kill_board_test_4.py:100]: Same

**Review by cbrxyz — APPROVED** [review]
Looks good, thanks!

**Pull request merged**